### ISSUE TITLE
Transform violates spec when slices are used

### ISSUE THREAD

**Issue by feliperoos** [3MFConsortium/lib3mf#59]
Slice Extension section 1.2 defines
"
[...]
Transforms: If an object references slice model data, the 3D transform matrices in
<build>/<item> and <component> elements are limited to those that do not impact the slicing
orientation (“planar transformations”). Therefore, any transform applied (directly or indirectly)
to an object that references an <slicestack> MUST have m02, m12, m20, and m21 equal to zero
and m22 equal to one
[...]"

In addition, it states:
"
1. Producers MUST NOT output other transformation string values than “0”, “0.” or “0.0”
with an arbitrary amount of trailing zeros for m02, m12, m20, and m21.
2. Producers MUST NOT output other transformation string values than “1”, “1.” or “1.0”
with an arbitrary amount of trailing zeros for m22.
"

Today, lib3mf is ignoring this rules and writing the values as they are in the transform.

**Comment by feliperoos** [comment]
@martinweismann What would be the correct behavior here? override the values or throw an error?

**Comment by martinweismann** [comment]
Hi @feliperoos , I suggest to raise an exception on export.

FYI:
Some classes(CModelMeshObject, CModelComponentsObject, and CModelBuildItem) already have a function "isValidForSlices" this might be useful for you.

**Comment by martinweismann** [comment]
I would follow the references from each build-item, or components-object (potentially via other components-objects) and check, whether the additional transformation on each reference is valid for slices.
If any target of such a reference-chain has a slicestack attached to it, you can check whether all transformations that lead to that target are valid for slices. If not, you found the violating transformation on a slicestack object.
(paraphrased from https://3mf.io/wp-content/uploads/2016/07/3MFsliceSpec.pdf , p11, "The \<build\> element has a single change for slice data")

**Comment by feliperoos** [comment]
All right, I've implemented the behavior on #66 and based on your comment from the spec implemented #68 
It would be great if you could take a look on those.

**Comment by martinweismann** [comment]
Hi @feliperoos 
thanks for your contributions.
While looking over it, I found a bigger problem in the implementation of the slice-extension: the current implementation does not allow slices on components-objects (which is allowed according to spec). https://github.com/3MFConsortium/lib3mf/pull/72 will fix this.

I would like to ask you to modify the code in your PR accordingly once https://github.com/3MFConsortium/lib3mf/pull/72 has been merged.
